File: src/main/kotlin/aurocosh/divinefavor/common/block/medium/MediumStone.java
package aurocosh.divinefavor.common.block.medium;

import net.minecraft.util.IStringSerializable;

public enum MediumStone implements IStringSerializable {
    NONE("none"),
    ARBOW("arbow"),
    BLIZRABI("blizrabi"),
    ENDERERER("endererer"),
    LOON("loon"),
    NEBLAZE("neblaze"),
    REDWIND("redwind"),
    ROMOL("romol"),
    SQUAREFURY("squarefury"),
    TIMBER("timber");

    // Optimization
    public static final MediumStone[] VALUES = MediumStone.values();

    private final String name;

    MediumStone(String name) {
        this.name = name;
    }

    @Override
    public String getName() {
        return name;
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/block/medium/MediumStone.kt
package aurocosh.divinefavor.common.block.medium

import aurocosh.divinefavor.common.lib.EnumIndexer
import aurocosh.divinefavor.common.lib.IIndexedEnum
import net.minecraft.util.IStringSerializable

enum class MediumStone(private val value: String) : IStringSerializable {
    NONE("none"),
    ARBOW("arbow"),
    BLIZRABI("blizrabi"),
    ENDERERER("endererer"),
    LOON("loon"),
    NEBLAZE("neblaze"),
    REDWIND("redwind"),
    ROMOL("romol"),
    SQUAREFURY("squarefury"),
    TIMBER("timber");

    override fun getName(): String {
        return value
    }

    companion object : IIndexedEnum<MediumStone> {
        override val indexer: EnumIndexer<MediumStone> = EnumIndexer(values())
    }
}